src/app/services/errorHandler/defaultErrorHandler.ts

```diff
@@ -1,13 +1,18 @@
-import { ErrorHandler, Injectable} from '@angular/core';
+import { ErrorHandler, Injectable, Injector, NgZone} from '@angular/core';
+import { Router } from '@angular/router';
 import { LoggerService } from '../logger/logger.service';
 
 @Injectable()
 export class DefaultErrorHandler implements ErrorHandler {
-  constructor(
-    private loggerService: LoggerService
-  ) { }
+   private router: Router;
+   constructor(
+     private loggerService: LoggerService,
+     private injector: Injector,
+     private zone: NgZone) { }
 
-    handleError(error: Error) {
-        this.loggerService.error(error);
+   public handleError(error: Error) {
+      this.loggerService.error(error);
+      this.router =  this.injector.get(Router);
```

Comment:
Out of curiosity Why does this need NgZone?

Reply:
Again Out of curiosity we could inject Router rather than explicitly using injector. Whats the need for it?

Reply:
> Again Out of curiosity we could inject Router rather than explicitly using injector. Whats the need for it?

If you inject the router is causes cyclic referencing error

Reply:
> Out of curiosity Why does this need NgZone?

NgZone has to be used because The errors in the error handlers are out of angular zone.
https://stackoverflow.com/questions/48787669/angular-navigation-in-errorhandler-only-works-with-zone-run